# Notebook 4: Text Preprocessing
In this notebook, we will import our modular preprocessing function from `src/preprocessing.py`, clean the job posting text fields, combine them into a single cleaned text field, and save the processed dataset for modeling.

In [1]:
import pandas as pd
import os
from src.preprocessing import clean_text

# Load raw dataset
data_path = os.path.join("..", "data", "fake_job_postings.csv")
df = pd.read_csv(data_path)

print("Raw shape:", df.shape)

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\alokk\AppData\Roaming\nltk_data...


[nltk_data]   Unzipping corpora\stopwords.zip.


[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\alokk\AppData\Roaming\nltk_data...


Raw shape: (17880, 18)


## 1. Combining Text Columns
We will fill nulls with empty strings and combine `title`, `company_profile`, `description`, `requirements`, and `benefits` into a single field `text`.

In [2]:
text_cols = ['title', 'company_profile', 'description', 'requirements', 'benefits']
for col in text_cols:
    df[col] = df[col].fillna('')

# Combine columns
df['combined_text'] = df['title'] + " " + df['company_profile'] + " " + df['description'] + " " + df['requirements'] + " " + df['benefits']
print("Sample combined text (first 100 chars):\n", df['combined_text'].iloc[0][:100])

Sample combined text (first 100 chars):
 Marketing Intern We're Food52, and we've created a groundbreaking and award-winning cooking site. We


## 2. Running Preprocessing
Now we apply our cleaning function to all rows. Because there are 17,880 rows, we will print progress.

In [3]:
print("Starting text cleaning (this may take a minute)...")
df['clean_text'] = df['combined_text'].apply(clean_text)
print("Finished cleaning!")
print("Sample cleaned text (first 100 chars):\n", df['clean_text'].iloc[0][:100])

Starting text cleaning (this may take a minute)...


Finished cleaning!
Sample cleaned text (first 100 chars):
 marketing intern food created groundbreaking award winning cooking site support connect celebrate ho


## 3. Saving the Processed Dataset
We keep only the columns relevant for modeling: `clean_text`, `telecommuting`, `has_company_logo`, `has_questions`, and the target `fraudulent`, then save to `data/processed_jobs.csv`.

In [4]:
processed_df = df[['clean_text', 'telecommuting', 'has_company_logo', 'has_questions', 'fraudulent']]

# Drop rows where clean_text is completely empty after cleaning
processed_df = processed_df[processed_df['clean_text'].str.strip() != '']

output_path = os.path.join("..", "data", "processed_jobs.csv")
processed_df.to_csv(output_path, index=False)
print("Saved processed data to:", output_path)
print("Processed shape:", processed_df.shape)

Saved processed data to: ..\data\processed_jobs.csv
Processed shape: (17880, 5)
